fix(userprofile): Update User Profile design

mindreeper2420 opened the pull request:
Update the design of the User Profile page.

related to https://github.com/fabric8-ui/fabric8-ux/issues/715

closes https://github.com/fabric8-ui/fabric8-ux/issues/731

![profileopenshift io](https://user-images.githubusercontent.com/4032718/32449877-b72f3ad4-c2e0-11e7-9cf3-7997e183d4c1.png)

fabric8cd commented:
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-2206-1 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2206-fabric8-ui.badger.fabric8.io

dlabrecq commented on src/app/profile/overview/activity/activity.component.html:
```diff
@@ -1,33 +0,0 @@
-      </li>
-    </ul>
-  </div>
-</ng-template>
```
Should we delete the activity component? I realize this isn't much, but we will show activity at some point?

mindreeper2420 replied:
I deleted it as there isn't anything in the pipeline right now for activity. It is just an empty component that wouldn't be rendered anywhere. I can put it back, but don't see the need right now.

dlabrecq commented on src/app/profile/overview/connected-accounts/connected-accounts.component.ts:
```diff
@@ -0,0 +1,71 @@
+      .switchMap(() => this.auth.openShiftToken)
+      .map(token => {
+        this.openShiftLinked = (token !== undefined && token.length !== 0);
+      });
```
This code isn't valid because you must to call map on loggedInUser before switchMap -- it will never step into the map where openShiftLinked is set. 

It's also unnecessary because openShiftLinked is already obtained via the constructor.

dlabrecq commented on src/app/profile/overview/connected-accounts/connected-accounts.component.ts:
```diff
@@ -0,0 +1,71 @@
+      this.gitHubLinked = (token !== undefined && token.length !== 0);
+    }));
+    // for OpenShift
+    this.contextSubscription = this.contexts.current.subscribe(val => {
```
You can use this.subscriptions.push(...) instead of this.contextSubscription.

mindreeper2420 replied:
I believe that is the next line - would that mean that I only need one? And then how would I call the `{{contextUsername}}` to display the OpenShift username?

dlabrecq replied:
You just don't need a contextSubscription variable. The code would look more like this:

this.subscriptions.push(this.contexts.current.subscribe(val => {
  this.contextUserName = val.user.attributes.username;
});

mindreeper2420 replied:
should be all set now

dlabrecq commented on src/app/profile/overview/connected-accounts/connected-accounts.component.ts:
```diff
@@ -0,0 +1,71 @@
+  subscriptions: Subscription[] = [];
+  contextSubscription: Subscription;
+
+  authGitHub: boolean = false;
```
loggedInUser & authGitHub appear to be unused?

dlabrecq commented on src/app/profile/overview/spaces/spaces.component.ts:
```diff
@@ -76,31 +76,6 @@ export class SpacesComponent implements OnDestroy, OnInit  {
-    this.spaceToDelete = space;
-    this.modalRef = this.modalService.show(deleteSpace, {class: 'modal-lg'});
-  }
-
```
It's unfortunate that users cannot delete their own spaces from this list. This means a user has to navigate away from this page and must "view all spaces".

mindreeper2420 replied:
Agreed, but that will be in future implementations. This is just the simplified version for the time being.

dlabrecq replied:
If the intention is to use this in the future, then I suggest leaving the code or comment it out for now. It took someone a bit of time to figure out how to delete a space.

mindreeper2420 replied:
added back, but commented out

dlabrecq commented on src/app/profile/overview/spaces/spaces.component.ts:
```diff
@@ -19,7 +19,7 @@ export class SpacesComponent implements OnDestroy, OnInit  {
   pageSize: number = 20;
   subscriptions: Subscription[] = [];
-  spaceToDelete: Space;
+  // spaceToDelete: Space;
```
Instead of a comment, we should just remove this line.

mindreeper2420 replied:
I left this line so that it matches with the delete space code that was also commented out.

dlabrecq commented on src/app/profile/overview/work-items/work-items.component.html:
```diff
@@ -50,3 +105,4 @@
     </ul>
   </div>
 </ng-template>
+ -->
```
You can also delete the commented-out code.

A return char is also needed at the end of file.

dlabrecq commented on src/app/profile/overview/work-items/work-items.component.less:
```diff
@@ -5,3 +5,7 @@
+.work-item-dropdown {
+  position: absolute;
+  right: 30px;
+}
```
Return char needed

dlabrecq commented on src/assets/stylesheets/shared/_overrides.less:
```diff
@@ -21,3 +21,22 @@
+    background-color: @color-pf-black-200 !important;
+    /* stylelint-enable */
+  }
+}
```
Return char needed

fabric8cd commented:
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-2206-2 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2206-fabric8-ui.badger.fabric8.io

joshuawilson commented:
@mindreeper2420 this looks like 4-5 fixes in one PR. Please refactor to be just one thing, then submit PRs for the others.

mindreeper2420 commented:
@joshuawilson all of the file changes here are for this PR. I’ve removed the changes that are separated submissions. There was a lot that had to be moved and changed.

joshuawilson commented:
ok, thanks

mindreeper2420 commented:
@joshuawilson rebased and synced with latest version of master.

fabric8cd commented:
@mindreeper2420 fabric8/fabric8-ui:SNAPSHOT-PR-2206-4 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2206-fabric8-ui.badger.fabric8.io
